### DISCUSSION TITLE
LocalStorage and Cache clear at initial load in nextjs

### DISCUSSION THREAD

**Discussion by golden-hero** [community/community#58892] [Programming Help]
### Body

In our project we need to give continuous updates, after updating the system, users need to clear their cache and localStorage data manually so that they can use the updates.

Is there any way to deal with the problem, where after updates all the cache will be cleared for once when the user comes to the site? the project is built upon Nextjs.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Beargruug** [comment]
Hey here is something you can do,

**localStorage Check**: You can use **localStorage** to store the version number of your application. On each update, you can increment the version number. When the user visits the site, you can compare the stored version with the current version. If there's a mismatch, it means there's a new version available, and you can programmatically clear the **localStorage** to ensure users get the latest data.

Here is a basic example my memory: 

```
// pages/_app.js

import { useEffect } from 'react';

function MyApp({ Component, pageProps }) {
  useEffect(() => {
    const currentVersion = '1.0.0'; // Set the current version of your app here

    const storedVersion = localStorage.getItem('appVersion');
    if (storedVersion !== currentVersion) {
      // Clear localStorage and update the stored version
      localStorage.clear();
      localStorage.setItem('appVersion', currentVersion);
    }
  }, []);

  return <Component {...pageProps} />;
}

export default MyApp;
```

**Comment by arunned** [comment]
thanks. This helped me solve an issue..!
